### PR TITLE
BUG: rendering dt64tz values with non-pytz

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/tslibs/vectorized.pyx [hunk 1]
```diff
@@ -136,6 +135,7 @@ def ints_to_pydatetime(
         tzinfo new_tz
         bint use_date = False, use_ts = False, use_pydt = False
         object res_val
+        bint fold = 0
```
Comment (mroeschke) [inline pandas/_libs/tslibs/vectorized.pyx:138]: So fold never needs to be set to 1 for the repr?

Also `fold` in the docstring can be removed

Comment (jbrockmendel) [reply pandas/_libs/tslibs/vectorized.pyx:138]: > So fold never needs to be set to 1 for the repr?

it doesn't matter in pytz cases and it gets set via pointer in non-pytz cases

> Also fold in the docstring can be removed

Will do in follow-up if thats OK

Comment (mroeschke) [reply pandas/_libs/tslibs/vectorized.pyx:138]: Yeah follow up PR is good

**Review by mroeschke — APPROVED** [review]

**Comment by rhshadrach** [comment]
This patch may have induced a potential regression. Please check the links below. If any ASVs are parameterized, the combinations of parameters that a regression has been detected appear as subbullets. This is a partially automated message.

 - https://asv-runner.github.io/asv-collection/pandas/#tslibs.tslib.TimeIntsToPydatetime.time_ints_to_pydatetime
   - box='datetime'; size=100; tz=<DstTzInfo 'US/Pacific' LMT-1 day, 16:07:00 STD>
   - box='datetime'; size=100; tz=tzfile('/usr/share/zoneinfo/Asia/Tokyo')
   - box='datetime'; size=10000; tz=<DstTzInfo 'US/Pacific' LMT-1 day, 16:07:00 STD>
   - box='datetime'; size=10000; tz=tzfile('/usr/share/zoneinfo/Asia/Tokyo')
   - box='datetime'; size=1000000; tz=<DstTzInfo 'US/Pacific' LMT-1 day, 16:07:00 STD>
   - box='datetime'; size=1000000; tz=tzfile('/usr/share/zoneinfo/Asia/Tokyo')
   - box='time'; size=100; tz=<DstTzInfo 'US/Pacific' LMT-1 day, 16:07:00 STD>
   - box='time'; size=100; tz=tzfile('/usr/share/zoneinfo/Asia/Tokyo')
   - box='time'; size=10000; tz=<DstTzInfo 'US/Pacific' LMT-1 day, 16:07:00 STD>
   - box='time'; size=10000; tz=tzfile('/usr/share/zoneinfo/Asia/Tokyo')
   - box='time'; size=1000000; tz=<DstTzInfo 'US/Pacific' LMT-1 day, 16:07:00 STD>
   - box='time'; size=1000000; tz=tzfile('/usr/share/zoneinfo/Asia/Tokyo')
   - box='timestamp'; size=100; tz=<DstTzInfo 'US/Pacific' LMT-1 day, 16:07:00 STD>
   - box='timestamp'; size=100; tz=tzfile('/usr/share/zoneinfo/Asia/Tokyo')
   - box='timestamp'; size=10000; tz=<DstTzInfo 'US/Pacific' LMT-1 day, 16:07:00 STD>
   - box='timestamp'; size=10000; tz=tzfile('/usr/share/zoneinfo/Asia/Tokyo')
   - box='timestamp'; size=1000000; tz=<DstTzInfo 'US/Pacific' LMT-1 day, 16:07:00 STD>
   - box='timestamp'; size=1000000; tz=tzfile('/usr/share/zoneinfo/Asia/Tokyo')

**Pull request merged**